docs: promql compatibility list (#162)
GreptimeTeam/docs: docs/user-guide/promql.md

# PromQL

## Introduction

> PromQL is the query language for Prometheus. It is a powerful and flexible language that can be used to retrieve data for alerting, graphing, and analysis.

Reviewer: ```suggestion
PromQL is the query language for Prometheus. It is a powerful and flexible language to retrieve data for alerting, graphing, and analysis.
```

> GreptimeDB has reimplemented PromQL in Rust, here is the compatibility list.

Reviewer: ```suggestion
GreptimeDB has reimplemented PromQL in Rust, here attaches the compatibility list.
```

Reviewer 2: ```suggestion
GreptimeDB has reimplemented PromQL Lexer and Parser in Rust, here is the compatibility list.
```

## Compatibility List

### Overview

- Basics
  - [Literal](#literal)
  - [Selector](#selector)
  - Subquery (unsupport)
- Operators
  - [Binary Operators](#binary-operators)
  - Vector Matching (unsupport)
  - [Aggregation Operators](#aggregators)
- Funtions
  - [Instant Functions](#instant-functions)
  - [Range Functions](#range-functions)
  - Others (unsupport)

### Literal

Both string and float literals are supported, with the same [rule](https://prometheus.io/docs/prometheus/latest/querying/basics/#literals) as PromQL.

### Selector

> Both instant and range selector are supported. The only unsupported exception is the label matching on metric name, e.g.: `__name__!="request_count"` (but the equal-matching case is supported: `__name__="request_count"`). Time duration and offset are supported, but `@` modifier is not supported yet.

Reviewer: ```suggestion
Both instant and range selectors are supported. The only exception is the label matching on metric name, e.g.: `__name__!="request_count"` (the equal-matching case is supported: `__name__="request_count"`). 
Time duration and offset are supported, but `@` modifier is not supported yet.
```

# PromQL

## Introduction

PromQL is the query language for Prometheus. It is a powerful and flexible language that can be used to retrieve data for alerting, graphing, and analysis.

GreptimeDB has reimplemented PromQL in Rust, here is the compatibility list.

## Compatibility List

### Overview

- Basics
  - [Literal](#literal)
  - [Selector](#selector)
  - Subquery (unsupport)
- Operators
  - [Binary Operators](#binary-operators)

>   - Vector Matching (unsupport)

Reviewer 2: Is it needed to use one of `unsupport` or `unsupported`?

Author: Looks "unsupport" is a wrong word, I've changed them to unsupported

  - [Aggregation Operators](#aggregators)
- Funtions
  - [Instant Functions](#instant-functions)
  - [Range Functions](#range-functions)
  - Others (unsupport)

### Literal

Both string and float literals are supported, with the same [rule](https://prometheus.io/docs/prometheus/latest/querying/basics/#literals) as PromQL.

### Selector

> Both instant and range selector are supported. The only unsupported exception is the label matching on metric name, e.g.: `__name__!="request_count"` (but the equal-matching case is supported: `__name__="request_count"`). Time duration and offset are supported, but `@` modifier is not supported yet.

Reviewer 2: ```suggestion
Both instant and range selector are supported. The only unsupported exception is the label matching on metric name, e.g.: `{__name__!="request_count}"` (but the equal-matching case is supported: `{__name__="request_count}"`). Time duration and offset are supported, but `@` modifier is not supported yet.
```

### Binary 

> *Pure literal binary-expr like `1+1` is not supported yet. `promql-parser` has some issues about grammar conflict between binary-expr and unary-expr.*

Reviewer 2: The conflict has been fixed

    | asin     | `asin(metric)`  |
    | atan     | `atan(metric)`  |
    | sin      | `sin(metric)`   |
    | cos      | `cos(metric)`   |
    | tan      | `tan(metric)`   |

- Unsupported:
    | Function                   | Progress |
    | :------------------------- | :------- |
    | absent                     | TBD      |
    | scalar                     | TBD      |
    | sgn                        | TBD      |
    | sort                       | TBD      |
    | sort_desc                  | TBD      |
    | timestamp                  | TBD      |
    | acosh                      | TBD      |
    | asinh                      | TBD      |
    | atanh                      | TBD      |
    | sinh                       | TBD      |
    | cosh                       | TBD      |
    | tanh                       | TBD      |
    | deg                        | TBD      |
    | rad                        | TBD      |
    | *other multiple input fns* | TBD      |

### Range Functions

- Supported:
    | Function | Example                |
    | :------- | :--------------------- |
    | increase | `increase(metric[5m])` |
    | irate    | `irate(metric[5m])`    |

>     | idelta   | `idelta(metric[5m])`   |

Reviewer 2: Maybe we can just leave the supported functions here? Some functions have not been listed here, like: [`<aggregation>_over_time()`](https://prometheus.io/docs/prometheus/latest/querying/functions/#aggregation_over_time).

Author: Updated
